glean-core/ios/GleanTests/GleanTests.swift

```diff
@@ -326,8 +319,26 @@ class GleanTests: XCTestCase {
         // in the main process
         Glean.shared.isMainProcess = false
 
-        // Restart glean
+        expectation = expectation(description: "GleanTests: Ping Received")
+        // We are using OHHTTPStubs combined with an XCTestExpectation in order to capture
+        // outgoing network requests and prevent actual requests being made from tests.
+        stubServerReceive { _, _ in
+            // Fulfill test's expectation once we parsed the incoming data.
+            DispatchQueue.main.async {
+                // Let the response get processed before we mark the expectation fulfilled
+                self.expectation?.fulfill()
+            }
+        }
+
+        // Invert the expectation so that it will assert if it gets fulfilled. Since Glean
+        // is simulating being initialized on another process, we should not get any pings
+        // since init should fail.
+        expectation?.isInverted = true
+        // Restart Glean
         Glean.shared.resetGlean(clearStores: false)
+        waitForExpectations(timeout: 2.0) { error in
+            XCTAssertNil(error, "Error while waiting for inverted expectation: \(error!)")
```

Comment:
Maybe make this more tailored to the test, e.g.:

```suggestion
            XCTAssertNil(error, "Received a ping upload when we shouldn't: \(error!)")
```